# Shewanella_genomics
Comparative genome analysis of the genus Shewanella unravels the association of key genetic traits with known and potential pathogenic lineages 

Gabriela N. Cerbino, German M. Traglia, Teolincacihuatl Ayala Nuñez, Gisela Parmeciano Di Noto, María Soledad Ramírez, Daniela Centrón, Andrés Iriarte*, Cecilia Quiroga*.


## Methods

## GO term enrichment analysis

## Workflow:

1. Download genomes 

2.EggNOG mapper

3.run_func.sh (input= faa file (proteome from each genome)- FASTA file with Amino acid sequence, output= Eggnog output files from each faa file)

4.fisher3.R (input=matrix file - output = table_fisher.csv (examples in folder: clin1_v_env, clin2_v_env, etc))

5.filter_signi.sh (input=table_fisher.csv, output=sign_pvalue_assign_func_c1_v_c2.tab (examples in folder: clin1_v_env, clin2_v_env, etc))

Conservation:

1 extract_seq.sh (KatG,hemB, etc)

2 GOATOOLS

3 blastp: all-to all of each amino acid protein

4 PROSITE & Pfam (ONLINE webserver)

– Code list:

The function of each script is described. The details of how to use each "script" are commented within the code.

## Scripts

$ count_GO_genomes_2.sh
Read and count each GO term on the functional annotation generated by EggNog Mapper v2 on a single genome.

$ run_func.sh
It fulfills a function similar to the script "count_GO_genomes_2.sh" but it does so on a collection of genomes that have been functionally annotated using EggNog. The output is a matrix of presence/absence of the GO and each of the included genomes.

$ fisher3.R
Performs the Fisher test from an MxN matrix with the following columns: “Type Functional Term [for example: GO]”; "number term without zeros"; "cond1_Presence"; “cond2_Presence_Term”; "cond1_Absence_Term"; “cond2_Absense_Term”.

$ filter_signi.sh
Extracts the category or level of the GO terms that were significant from the Fisher test.

$ extract_seq.sh
Extract a sequence or list of amino acid sequences from a fasta file

## Command lines of each program.
 --EggNOG Mapper v2. 

The functional annotation of each of the genomes was performed using the following command line:

$ emapper.py -i FASTA_FILE_PROTS -o test --tax_scope bacteria

 -- GOATOOLS:
We complement the annotation of the results obtained with the script "filter_signi.sh" with one of the tools of the GOATOOLS package.

$ go_plot.py --go_file=tests/data/go_plot/go_selected.txt -r -o selected_go.png

 --PROSITE y Pfam:
We use the online servers of each database to perform the blastp.




## Phylogenetic analysis

 ## Workflow:

Download protein sequences —>  get_homologues.pl  —> Clustal Omega v1.2.0
|
|
IQ-Tree v1.6.12 <— concatenador.sh

## Code list:

 -- Concatenation of alignment of orthologous genes. concatenador.sh takes all the alignments of orthologs made by clustalo software, which were first identified by get_homologues.pl. Then the “super” protein was used in IQ-tree for phylogenetic analysis, including model prediction and bootstrap analysis. concatenador.sh run within the same folder where all alignments (.aln) files are located. Note that the default output of clustalo is an alignment in fasta format with a “.aln” extension..  

$ ./concatenador.sh

